lib/stack.rb

```diff
@@ -1,19 +1,21 @@
+require_relative './linked_list.rb'
+
 class Stack
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = LinkedList.new
   end
 
   def push(element)
-    raise NotImplementedError, "Not yet implemented"
+    @store.add_last(element)
+    return @store
   end
 
   def pop
-    raise NotImplementedError, "Not yet implemented"
+    return @store.remove_last()
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return true if @store.get_last.nil?
```

Comment:
LinkedList also has an `empty?` method.

```suggestion
    return @store.empty?
```